pennylane/kernels/cost_functions.py

```diff
@@ -175,3 +176,59 @@ def circuit(x1, x2):
         rescale_class_labels=rescale_class_labels,
         normalize=True,
     )
+
+
+def task_model_alignment(
+    N,
+    task_weights,
+    kernel_evals,
+):
+    r"""Task-model alignment for a given kernel function as proposed in [].
+
+    This function measures how much of the target function producing a supervised dataset is
+    captured in the kernel's first :math:`N` eigenvalues. More colloquially, it measures how much the basis functions
+    available to the kernel are used by the target function that we seek to learn with the kernel.
+
+    Let :math:`w_k` be the task weights for the target function that produced a supervised dataset as specified
+    by the :doc:`pennylane.kernels.utils.task_weights` function, and :math:`\lambda_k` the eigenvalues of the kernel
+    used to compute these weights. The task-model alignment is defined as:
+
+    .. math::
+
+        C(l) = \frac{\sum_{k \leq N} w_k^2 \lambda_k}{\sum_{k} w_k^2 \lambda_k}.
+
+    Evidence put forward in Ref [] and other papers shows that this alignment is closely related to the generalization
+    error of a kernel method.
+
+    Args:
+        N (int): compute the task-model alignment for components up this index; has to be smaller or equal to the
+            number of data points
+
+    Returns:
+        float: The task-model alignment of the kernel on the dataset.
+
+    **Example:**
+
+    Consider a simple kernel function based on :class:`~.templates.embeddings.AngleEmbedding`:
+
+    .. code-block :: python
+
+        dev = qml.device('default.qubit', wires=2, shots=None)
+
+        @qml.qnode(dev)
+        def kernel(x1, x2):
+            qml.templates.AngleEmbedding(x1, wires=dev.wires)
+            qml.adjoint(qml.templates.AngleEmbedding)(x2, wires=dev.wires)
+            return qml.expval(qml.Projector([0, 0]), wires=[0, 1]))
+
+    We can then compute the task-model alignment on a set of 4 (random)
+    feature vectors ``X`` via
+
+    >>> X = np.random.random((4, 2))
+    >>> qml.kernels.task_model_alignment(X, kernel)
```

Comment:
this seems out of date when looking at the current function definition :thinking: